classes/Country.php

```diff
@@ -184,7 +184,7 @@ public static function getCountriesByIdShop($idShop, $idLang)
     public static function getByIso($isoCode, $active = false)
     {
         if (!Validate::isLanguageIsoCode($isoCode)) {
-            die(Tools::displayError());
+            die(Tools::displayError('Given iso code (' . $isoCode . ') is not valid.'));
```

Comment:
I see something similar in [classes/Carrier.php](https://github.com/PrestaShop/PrestaShop/blob/8.1.x/classes/Carrier.php#L616) and [classes/Cart.php](https://github.com/PrestaShop/PrestaShop/blob/8.1.x/classes/Cart.php#L1513). So I think we can apply similar PRs throughout the project.
On the other side, in [classes/Tools.php](https://github.com/PrestaShop/PrestaShop/blob/8.1.x/classes/Tools.php#L1106-L1128), `$errorMessage` is processed via `getTranslator()->trans`. Why don't we use similar `trans` in this PR?

Reply:
@leemyongpakvn Maybe we can replace all occurences with `throw new PrestaShopException('the message');`?

Reply:
@Hlavtox Yes, the clearer message the better, but it is over the scope of this PR ;) And I'm worried about the performance when adding a bunch of new translation strings too.

Reply:
@leemyongpakvn don't worry about the performance, first of all, those are not translatable phrases, but adding those won't hurt the performance ;-)

Reply:
@kpodemski I see a [Merged PR](https://github.com/PrestaShop/PrestaShop/commit/6d87578a152fe9bd409788089f1b52a02b5740b2) derived from @matthieu-rolland, that _Generic error messages_. Maybe we need an ADR to avoid going back and forth between Explicit and Generic error messages ?

Reply:
@leemyongpakvn @kpodemski So guys, can I do it? There are only about 80 calls for `displayError`, so it's not that much work.

I would replace everything with exceptions and put some nicer error messages there. And we can stop using that function.

Reply:
@Hlavtox 

I guess there was a reason why somebody decided to display this instead of PrestaShopException, only if you are in developer mode, it shows you more errors. My understanding is that if we don't use `displayError` you will only see `Error 500` or a full exception message in the developer mode. With `displayError` you see some messages and exceptions in the developer mode.

@leemyongpakvn I must admit I don't know the reason for the change that @matthieu-rolland submitted, it was probably related to some security fix not to share a specific hook name.

To be honest, I think we discuss something that's different from the original PR. Original PR tried to add _some message_ instead of having "FatalError", which is a good thing. Maybe we should focus on that, so replace `displayError()` with a more meaningful message.

Reply:
@Hlavtox I agree with your suggestion, but council members will decide to do it or not ;)

Reply:
@leemyongpakvn Thank god I am also a member!